src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/RazorSourceGeneratorComponentTests.cs

```diff
@@ -713,4 +713,44 @@ public async Task LineMapping_Tabs()
             // Assert.Equal(new TextSpan(originalIndex, snippet.Length), mappedSpan);
         }
     }
+
+    [Fact, WorkItem("https://github.com/dotnet/razor/issues/9381")]
+    public async Task UnrecognizedComponentName()
+    {
+        // Arrange
+        var project = CreateTestProject(new()
+        {
+            ["Shared/Component1.razor"] = """
+                <X1 />
+                <X2 @key="null" />
+                <X3 @ref="x" />
+                <X4 @bind="x" />
+                <X5 @bind-Value="x" @bind-Value:event="oninput" />
+                <X6 @formname="n" />
+                <X7 @rendermode="null" />
+
+                @code {
+                    object? x;
+                }
+                """,
+        });
+        var compilation = await project.GetCompilationAsync();
+        var driver = await GetDriverAsync(project);
+
+        // Act
+        var result = RunGenerator(compilation!, ref driver);
+
+        // Assert
+        result.Diagnostics.Verify(
```

Comment:
Let's prefer `VerifyDiagnostics`. I'd like to see the full roslyn-style "here's what this error is and what syntax it's being reported on" comments.

Reply:
I'm not sure what you mean, I'm not aware of any other `VerifyDiagnostics` than the `Diagnostics.Verify` I'm using.

It's not reporting the syntax because the underlying `Diagnostic`s are not `CSharpDiagnostic`s as required by the utility:

https://github.com/dotnet/roslyn/blob/c5d20779f421caafbd4e18e3bfc5c0320a0a29f8/src/Compilers/Test/Core/Diagnostics/DiagnosticDescription.cs#L490-L491

Perhaps that could be changed on the roslyn side, though? I would also very much like to see the syntax and error message.

Reply:
Here it's simple though - apart from those diagnostics that I added comment to, every column number is 1 and row number corresponds to the component name (`X1`, `X2`, ...) so it should be clear what every diagnostic belongs to.

Reply:
Ah, I see. We may want to update that to give us the error messages for razor diagnostics as well then.

Reply:
https://github.com/dotnet/roslyn/pull/70691